src/tactic/nat_cases.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Scott Morrison
+
+Case bashing on natural numbers.
+
+`nat_cases a ≤ n < b` breaks into the following cases:
+`n < a`, one case `n = k` for each `a ≤ k < b`, and `n ≥ b`
+
+`nat_cases n`
+1) inspects hypotheses looking for "easy" upper and lower bounds on `n`,
+2) calls `nat_cases a ≤ n < b` with appropriate values of `a` and `b`,
+3) calls `linarith` to discharge the `n < a` and `n ≥ b` cases.
+-/
+import tactic.fin_cases
+import tactic.linarith
+import data.nat.basic
+
+open list
+
+/--
+This is true reflexively, by the definition of `≤` on ℕ, but it's still useful to have,
+to convince Lean to change the syntactic type.
+-/
+lemma nat.add_one_le_iff {a b : ℕ} : a + 1 ≤ b ↔ a < b := iff.refl _
+
+/--
+For any natural numbers n, a, and b, one of the following holds:
+1. n < a
+2. n ≥ b
+3. n ∈ Ico a b
+-/
+lemma nat.Ico_trichotomy (n a b : ℕ) : n < a ∨ n ≥ b ∨ n ∈ Ico a b :=
```

Comment:
Should these non-meta lemmas move to non-`tactic/` parts of the library?

Reply:
Done.